### ISSUE TITLE
Submission 3 (LearnExp/Short)

### ISSUE THREAD

**Issue by opencert** [opencert/workshop-2019#3]
Title: Open Source Software as a Learning Tool for Computer Science Students

Abstract:
In this paper student experience of contributing to Open Source Software (OSS) is described. Contributions were done as a part of the OSS course taken at Nazarbayev University during Spring 2019 term. Two students described their experience: Assiya Kuzyakhmetova and Aidarbek Suleimenov. Their motivations to contribute to OSS, selected projects and the lists of activities were also mentioned in this paper. Assessment of this experience is also done in this publication.
[OpenCERT_2019_paper_3.pdf](https://github.com/opencert/workshop-2019/files/3408428/OpenCERT_2019_paper_3.pdf)

**Comment by Stamelos** [comment]
I will review this submission.
Ioannis Stamelos

**Comment by goriR** [comment]
I would like to review this submission.

Roberta Gori

**Comment by opencert** [comment]
Assigned PC members:
- Ioannis Stamelos
- Roberta Gori
- Elsa Estevez
- Donatella Persico

**Comment by sksowe** [comment]
I would like to have a look at this paper

**Comment by goriR** [comment]
I find the part of Assiya very difficult to follow. In particular Section 2.1 The Technical Aspects and General Aspects of the project should be deeply revised in order to make better understand what the project aimed to do. These sections  should explain the aim of  the project and the kind of technologies used rather than just listing their names or acronyms.

**Comment by elsaestevez** [comment]
I will do the review

**Comment by Stamelos** [comment]
I agree with goriR comment. In general, the paper is too technical but it manages to deliver useful information, i.e. a detailed account of the experience of the two students and their (positive) opinion on students working/learning in OSS projects. Paper might be improved as follows:
1. Improve English, ask help from a senior English speaker
2. Insert a section on theoretical background analysis of students practicing and
learning with OSS. Provide a short list of references to relevant published works.
3. Building on (2): what has been reported up to now (about benefits, learning outcomes, issues and
how to address them, improvements) and whether the experience of the two students confirms or not such knowledge. 
4. The issue of late response to one student's queries worths further analysis: why it happened, was this attitude observed with regular contributors as well, should a more formal role be introduced in OSS (such as 'Student Learner'), would a  mentorship relation help, a.s.o.?

**Comment by elsaestevez** [comment]
The paper presents the first-hand experience from two students who participated in open source projects as part of a university course. For each of the projects, the students explain the project technical and general aspects, their motivation to select the project, the role they played and the type of interactions with the project community, the challenges they faced, and the type of activities and work that they performed. Finally, each student presents their assessment of the experience and synthesize some general conclusions.

Main strengths of the paper is documenting the first-hand experience of two students participating in open source projects aiming at teaching programming- and project management-related competencies to Computer Science students.

Main weaknesses of the paper refer to the lack of generalization of results and its scientific contribution. This could be improved with some discussion with related work and summarizing some recommendations for lecturers involved in similar experiences. 

Some minor details to improve the paper quality include: 1) revising the sentences to avoid passive voice, 2) including quotations for highlighting the student´s opinions specifically related to their experience, and 3) revising the use of capital and small letters, and punctuation marks.

**Comment by Donatella-Persico** [comment]
I have reviewed this paper. Here are my comments.
This paper reports on the experience of two computer science students (the authors) while learning by participating to the GitHub community. While the paper is reasonably well written and organized, its main problem is that it is not a research paper, but rather the account of two personal experiences, which cannot have any ambition of being generalizable. I see several ways the paper can be improved, listed in the following.
•	The authors should make an effort to identify and describe the aims of the paper and/or formulate explicitly their research questions. These could be “identify pros and cons of this approach to learn”, compare this approach to more “traditional ones”, or a combination of the two.
•	The authors should add a “state of the art” section where they summarize what we (the reserach community) already know about learning with this approach, citing relevant papers from the academic literature. This would also enrich the bibliography section, which is presently very poor, with most references not being research papers.
•	The authors should try to present their qualitative “data” (unfortunately, such data are only the opinions of the authors) in a way that they can be regarded as more objective. One possibility is, based on the authors experience/conclusions, prepare a survey and ask their colleagues to fill it in, in order to confirm or contradict their hypothesis. Another way is to introduce a third author who could interview the 2 students, analyse the interviews transcripts/recordings, and then present the emerged information (which could be very similar to those already provided in the paper) in the third person. 
•	The authors should not use the names of the students in the text of the paper, even if they coincide with the authors. They should anonymize the sources of the information reported. Ideally, the students’ reports should be written in the third person.
•	The authors should re-organise sections 2,3,5 around the research questions, so that the reader understands how the research questions are being answered by the available data.
•	The authors should have the English revised by a native speaker and should avoid including too many technical details about their projects, if these are not relevant to provide an answer to the research question(s).
I have also annotate the origical comment to provide more punctual suggestions, so I attach the annotated document.
Donatella
[OpenCERT_2019_paper_3_annotated.pdf](https://github.com/opencert/workshop-2019/files/3451291/OpenCERT_2019_paper_3_annotated.pdf)

**Comment by rhcu** [comment]
Thanks a lot for all your comments. We will try to address all of them in the revised version!
Regards,
Assiya Khuzyakhmetova

**Comment by rhcu** [comment]
@goriR 
Thanks for your comment. I will revise this section. 

However, I wanted to mention that I haven't listed names or acronyms of technologies but tried to explain what for they are used in the project. 

Can you elaborate more on your comment? Does it mean that you think that there are too many technical details?

**Comment by rhcu** [comment]
Attached to this comment you can find a revised version of our Learning Experience paper. 
[Khuzyakhmetova, Suleimenov, LearnExp.pdf](https://github.com/opencert/workshop-2019/files/3464338/Khuzyakhmetova.Suleimenov.LearnExp.pdf)

@Stamelos Thanks for your comments! Below you can find answers to them:
1) Thanks for pointing out. We rechecked the grammar to fix errors. 
2) We added some theoretical background and references. While the list is not so big, we will try our best to make it larger if we will be accepted before submitting the camera-ready version of the paper. 
3) The University of Scövde's experience with OSS course for Master's students was considered for that. 
4) We generalized the part about interaction with the community in the conclusion part and drawn out that it is important to choose a project that shows a quick response for contributors. A possible reason why it was not the case at one of the students was also considered and described in the conclusion.

@Donatella-Persico Thanks for your comments, and, especially, for the annotated file. It was very useful for us. We addressed the comments from the annotated file in the paper.  Answers:
1) We formulated a research question to be "If the freedom of choice in terms of OSS project and technology affects students' ability to contribute".  The same kind of question was considered in the paper describing an experience of the University of Scövde, and we tried to show that our conclusion corresponds to that question.
2) Added some background information.
3) We anonymized the paper with the use of Student A and Student B substitutions instead of names. 
4) Revised sections.
5) Rechecked the grammar and asked for proofreading. Removed some technical details.

@elsaestevez Thanks for your comments and for pointing out strengths and weaknesses of the paper! We tried to add more generalization in the revised version of the paper. We also updated the use of punctuation and capital letters. While it was not possible to remove all passive voice constructions from the paper at that point due to time constraints, we will try to improve this if we will be accepted before submitting the camera-ready version of the paper.

@goriR Thanks again for your comment. We updated Section 2 to decrease the number of technical details.

**Comment by Donatella-Persico** [comment]
Dear Authors,
thank you for your reply. Just a couple of additional suggestions in the attached file.
Donatella
[Opencert second version.pdf](https://github.com/opencert/workshop-2019/files/3468332/Opencert.second.version.pdf)

**Comment by aidarbek** [comment]
Dear @Donatella-Persico,

Thank you for your comments, please find the paper in the file attached.

[OSS_learning_experience.pdf](https://github.com/opencert/workshop-2019/files/3469105/OSS_learning_experience.pdf)

**Comment by opencert** [comment]
Camera-ready version for the post-proceedings.
[paper_3.pdf](https://github.com/opencert/workshop-2019/files/3918572/paper_3.pdf)

**Comment by opencert** [comment]
Camera-ready version for the post-proceedings.
[paper_4.pdf](https://github.com/opencert/workshop-2019/files/3918578/paper_4.pdf)
